.github/workflows/tools.yml

```diff
@@ -0,0 +1,56 @@
+name: "tools update"
+on:
+  schedule:
+    # Run once a week at 00:05 AM UTC on Saturday.
+    - cron: '5 0 * * 6'
+
+  workflow_dispatch:
+
+jobs:
+  tools_update:
+    if: github.repository == 'nodejs/node'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false  # Prevent other jobs from aborting if one fails
+      matrix:
+        include:
+          - id: eslint
+            run: |
+              cd tools
+              NEW_VERSION=$(npm view eslint dist-tags.latest)
+              CURRENT_VERSION=$(node -p "require('./node_modules/eslint/package.json').version")
+              if [ "$NEW_VERSION" != "$CURRENT_VERSION" ]; then
+                echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+                tools/update-eslint.sh
+              fi
+          - id: "@babel/eslint-parser"
+            run: |
+              cd tools
+              NEW_VERSION=$(npm view @babel/eslint-parser dist-tags.latest)
+              CURRENT_VERSION=$(node -p "require('./node_modules/@babel/eslint-parser/package.json').version")
+              if [ "$NEW_VERSION" != "$CURRENT_VERSION" ]; then
+                echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+                tools/update-babel-eslint.sh
+              fi
+          - id: "lint-md dependencies"
+            run: |
+              cd tools/lint-md
+              NEW_VERSION=$(npm outdated --parseable | cut -d: -f4 | xargs)
+              if [ "$NEW_VERSION" != "" ]; then
+                echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+                rm -rf package-lock.json node_modules && npm install --ignore-scripts)
+                make lint-md-rollup
+              fi
+    steps:
+      - uses: actions/checkout@v2
+      - run: ${{ matrix.run }}
+      - uses: gr2m/create-or-update-pull-request-action@v1  # Create a PR or update the Action's existing PR
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_USER_TOKEN }}
+        with:
+          author: Node.js GitHub Bot <github-bot@iojs.org>
+          body: "This is an automated update of ${{ matrix.id }} to ${{ env.NEW_VERSION }}."
```

Comment:
Personally I was thinking of something like
```md
This is an automated update of ${{ matrix.id }} and its dependencies. Updated:
`eslint@x.y.z` to `x.y.z1`
`eslint-plugin-markdown@x.y.z` to `x.y.z1`
```

```yml
matrix:
  include:
    - id: eslint
      packages: [eslint, eslint-plugin-markdown] # or '["eslint",...]', I don't know if Actions stringifies this
      before: cd tools
      update: tools/update-eslint.sh
```
```yml
  - run: |
      ${{ matrix.before }}
      NEW_VERSIONS=$(npm outdated --json | jq -r '.[] | select(.current != .wanted and .package as $p | ${{ matrix.packages }} | index($p)) | "`" + .package + "@" + .current + "` to `" + .wanted + "`"')
      if [ "$NEW_VERSIONS" != "" ]; then
        
        ${{ matrix.update }}
      fi

  - if: env.NEW_VERSIONS
    ...
    body: |
      This is an automated update of ${{ matrix.id }} and its dependencies. Updated:
      ${{ env.NEW_VERSIONS }}
```

Reply:
I like the idea, but there are several obstacles that make me want to consider this for a subsequent iteration:

1. `npm outdated` will not work for `eslint` or `@babel/eslint-parser` the way we have them installed. That's why I'm doing things the way I am in those instances.
2. I don't think the passing of a multiline value for the environment variable works in GitHub Actions. (I could be wrong about that, but that's what I seem to recall. If `$NEW_VERSIONS` contains newline characters, then `"NEW_VERSIONS=$NEW_VERSIONS"` won't work.)

For `lint-md`, the format you propose would be better than what I'm doing, as the way I'm doing it will result in lines longer than 72 characters frequently.

Reply:
> 2\. I don't think the passing of a multiline value for the environment variable works in GitHub Actions. (I could be wrong about that, but that's what I seem to recall. If `$NEW_VERSIONS` contains newline characters, then `"NEW_VERSIONS=$NEW_VERSIONS"` won't work.)

I suppose I can use some other character as a placeholder for the newline and then convert it in the relevant step.

Reply:
If the new versions are in a string with a format like "abc@1.2.3 xyz@4.5.6 foobar@42.0.0" then I can convert the spaces to newlines and the problem is solved.

Reply:
Indeed, but it is possible by using outputs and escaping new lines.  The runner will automatically take care of unescaping it:
```yml
  - run: |
      ...
        echo "::set-output name=version::${NEW_VERSIONS//$'\n'/'%0A'}"
      ...
    id: update
```
And then use it as `${{ steps.update.outputs.version }}`